vnet/lib/vnet/endpoints/1.0/topologies.rb

```diff
@@ -28,98 +28,70 @@ def self.put_post_shared_params
     update_by_uuid(:Topology)
   end
 
-  param_uuid M::Interface, :interface_uuid, required: true
-  post '/:uuid/datapaths/:datapath_uuid' do
-    topology = uuid_to_id(M::Topology, "uuid", "topology_id")
-    datapath = uuid_to_id(M::Datapath, "datapath_uuid", "datapath_id")
-    interface = uuid_to_id(M::Interface, "interface_uuid", "interface_id")
+  TOPOLOGY_ASSOCS = [
+    ['datapath', true],
+    ['network', nil],
+    ['segment', nil],
+    ['route_link', nil]
+  ].freeze
+
+  TOPOLOGY_ASSOCS.each { |other_name, with_interface|
+    other_id = "#{other_name}_id".to_sym
+    other_uuid = "#{other_name}_uuid".to_sym
+    assoc_table = "topology_#{other_name}s".to_sym
+
+    other_model = M.const_get(other_name.camelize)
+    assoc_model = M.const_get("Topology#{other_name.camelize}")
+    assoc_response = R.const_get("Topology#{other_name.camelize}")
+    
+    # TODO: Change to confirm with either POST or PUT idioms.
+    param_uuid M::Interface, :interface_uuid, required: true if with_interface
+    post "/:uuid/#{other_name}s/:#{other_uuid}" do
+      uuid_to_id(M::Topology, :uuid, :topology_id)
+      uuid_to_id(other_model, other_uuid, other_id)
+      uuid_to_id(M::Interface, :interface_uuid, :interface_id) if with_interface
+      
+      remove_system_parameters
+      respond_with(assoc_response.generate(assoc_model.create(params)))
+    end
+
+    get "/:uuid/#{other_name}s" do
+      show_relations(:Topology, assoc_table)
+    end
+
+    # TODO: No need to have other_uuid.
+
+    delete "/:uuid/#{other_name}s/:#{other_uuid}" do
+      uuid_to_id(M::Topology, :uuid, :topology_id)
+      other = uuid_to_id(other_model, other_uuid, other_id)
+
+      remove_system_parameters
+      assoc_model.destroy(params)
+
+      respond_with([other.uuid])
+    end
+  }
+
+  post '/:uuid/underlays/:underlay_uuid' do
```

Comment:
This new underlays relation needs to be tested in rspec with the other webapi routes.

Also at first sight I find it a little weird for underlays to be added to overlays instead of the other way around but that's not a big deal imo.